kubetest/e2e.go

```diff
@@ -393,6 +436,8 @@ type deployer interface {
 	IsUp() error
 	SetupKubecfg() error
 	Down() error
+	FederationUp() error
```

Comment:
I am not opposed to splitting this out as its own deployment. Just wondering about the pros and cons.

Do you see why making federation a separate deployment is better?